Global retrofit error interceptor

ammargitham opened the pull request:
Adding a global handler to intercept common errors and notify users. Still wip.

ammargitham commented:
I've added some error checks. We will need to add more for anonymous users specifically.

austinhuang0131 commented:
GraphQL errors:

* 200 `{"data":{"user":null},"status":"ok"}` => Bad user ID
* 500 `{"message":"execution failure","errors":[{"message":"execution error","locations":[]}],"data":{"user":null},"status":"fail"}` => JSON/query hash is valid, but cannot find (eg. bad after cursor)
* 400 `{"message":"(whatever)","status":"fail"}` => JSON or query hash failure

`?__a=1` errors:

* 404 `{}` => private or inexistent post, banned or inexistent hashtag, inexistent location
* Invalid usernames redirect to login.

ammargitham commented:
* 200 `{"data":{"user":null},"status":"ok"}` => Bad user ID
This should be handled inside profile fragment.

* 500 `{"message":"execution failure","errors":[{"message":"execution error","locations":[]}],"data":{"user":null},"status":"fail"}` => JSON/query hash is valid, but cannot find (eg. bad after cursor)
Added

* 400 `{"message":"(whatever)","status":"fail"}` => JSON or query hash failure
Added

`?__a=1` errors:

* 404 `{}` => private or inexistent post, banned or inexistent hashtag, inexistent location
Added
* Invalid usernames redirect to login.
Invalid usernames are giving 404 for me. Only after a rate limit, does it go to login page.
